lib/ansible/plugins/filter/mathstuff.py

```diff
@@ -45,6 +45,23 @@ def unique(a):
     return c
 
 
+def duplicate(a):
+    if isinstance(a, collections.Hashable):
+        c = set(a)
```

Comment:
But this is `unique` no?

I'd also add tests.

Reply:
I'm by no means a python expert, but I think line 49 and 50 means, if the argument is a dictionary, then return it without alteration.

As dictionary's aren't hashable, but lists are, and this filter only operates on lists, then this check needs to be made.

As I say I could be wrong on this though.

Ill try and find some time later to knock up a few tests...